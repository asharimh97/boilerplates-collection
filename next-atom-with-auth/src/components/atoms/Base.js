import styled from "styled-components";
import {
  compose,
  layout,
  position,
  space,
  color,
  flexbox,
  grid,
  background,
  border,
  shadow,
  typography
} from "styled-system";
import propTypes from "@styled-system/prop-types";

import theme from "styles/theme";

const Base = styled("div")(
  compose(
    background,
    border,
    color,
    flexbox,
    grid,
    layout,
    position,
    shadow,
    space,
    typography
  )
);

Base.displayName = "Base";
Base.defaultProps = {
  theme
};

Base.propTypes = {
  ...propTypes.background,
  ...propTypes.border,
  ...propTypes.color,
  ...propTypes.flexbox,
  ...propTypes.grid,
  ...propTypes.layout,
  ...propTypes.position,
  ...propTypes.shadow,
  ...propTypes.space,
  ...propTypes.typography
};

export default Base;
