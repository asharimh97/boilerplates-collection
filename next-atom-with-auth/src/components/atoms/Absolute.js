import styled from "styled-components";
import Base from "./Base";

const Absolute = styled(Base)`
  position: absolute;
`;

export default Absolute;
