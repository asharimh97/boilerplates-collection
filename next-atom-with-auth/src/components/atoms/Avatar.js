import styled, { css } from "styled-components";
import { convertNumberToPixel } from "utils/helper";
import Image from "./Image";

const sizes = {
  small: { width: "40px", height: "40px" },
  regular: { width: "50px", height: "50px" },
  big: { width: "65px", height: "65px" }
};

const Avatar = styled(Image).attrs(props => ({
  autofit: true,
  height: sizes[props.size]?.width || convertNumberToPixel(props.size),
  fullRounded: props.circle
}))`
  ${props =>
    props.size && !Number.isInteger(props.size)
      ? css`
          ${sizes[props.size]}
        `
      : css`
          width: ${props.size};
          height: ${props.size};
        `}
`;

Avatar.propTypes = {};

Avatar.displayName = "Avatar";

Avatar.defaultProps = {
  size: "regular",
  src: "/static/images/avatar.jpeg"
};

export default Avatar;
