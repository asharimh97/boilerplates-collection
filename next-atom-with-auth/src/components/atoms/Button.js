import { Button as AntButton } from "antd";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { flexbox, layout, space, typography } from "styled-system";
import propTypes from "@styled-system/prop-types";

const flex = props => props.flex && { display: "flex" };

const color = props => {
  if (props.color) {
    if (props.type === "link" || props.type === "text") {
      return css`
        &,
        &:hover,
        &:active,
        &:focus {
          color: ${props.theme.colors[props.color]?.["primary"] || props.color};
        }
      `;
    }

    return css`
      background-color: ${props.theme.colors[props.color]?.["primary"] ||
      props.color};
      border-color: ${props.theme.colors[props.color]?.["primary"] ||
      props.color};
      color: white;

      &:hover,
      &:active,
      &:focus {
        background-color: ${props.theme.colors[props.color]?.["60"] ||
        props.color};
        border-color: ${props.theme.colors[props.color]?.["60"] || props.color};
        color: white;
      }
    `;
  }

  return null;
};

const Button = styled(AntButton)`
  ${color};

  ${flex};
  ${flexbox};
  ${layout};
  ${space};
  ${typography}
`;

Button.propTypes = {
  flex: PropTypes.bool,
  ...propTypes.flexbox,
  ...propTypes.layout,
  ...propTypes.space,
  ...propTypes.typography
};

Button.displayName = "Button";

export default Button;
