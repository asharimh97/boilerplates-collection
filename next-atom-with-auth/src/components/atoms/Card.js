import styled from "styled-components";
import Base from "./Base";

const Card = styled(Base)``;

Card.defaultProps = {
  background: "white",
  p: 3,
  borderRadius: "5px",
  boxShadow: 5
};

Card.displayName = "Card";

export default Card;
