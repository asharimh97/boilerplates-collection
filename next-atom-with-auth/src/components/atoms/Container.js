import styled from "styled-components";
import Base from "./Base";

const flex = props => (props.flex ? { display: "flex" } : null);

const fluid = props => props.fluid && { maxWidth: "100%" };

const Container = styled(Base)`
  ${flex};
  ${fluid};
`;

Container.displayName = "Container";

Container.defaultProps = {
  mx: "auto",
  width: "100%",
  maxWidth: 640
};

export default Container;
