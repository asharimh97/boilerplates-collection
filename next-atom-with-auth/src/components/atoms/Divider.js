import styled, { css } from "styled-components";
import { borderColor, space } from "styled-system";
import Box from "./Box";

const Divider = styled(Box)`
  border: none;
  border-top: solid 1px;
  color: transparent;
  display: block;
  width: 100%;

  ${props =>
    props.vertical &&
    css`
      display: inline-block;
      border-top: none;
      border-left: solid 1px;
      font-size: inherit;
      margin: 0 12px;
      min-height: 0.9em;
      width: 1px;

      ${space};
    `}

  ${borderColor}
`;

Divider.defaultProps = {
  borderColor: "#ccc"
};

Divider.displayName = "Divider";

export default Divider;
