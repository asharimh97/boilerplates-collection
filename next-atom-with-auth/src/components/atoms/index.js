export { default as Absolute } from "./Absolute";
export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Button } from "./Button";
export { default as Card } from "./Card";
export { default as Container } from "./Container";
export { default as Divider } from "./Divider";
export { default as Fixed } from "./Fixed";
export { default as Flex } from "./Flex";
export { default as Grid } from "./Grid";
export { default as Image } from "./Image";
export { default as Link } from "./Link";
export { default as Sticky } from "./Sticky";
export { default as Text } from "./Text";
export { default as Title } from "./Title";
