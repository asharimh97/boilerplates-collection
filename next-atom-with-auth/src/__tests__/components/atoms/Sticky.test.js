import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { create } from "react-test-renderer";
import "jest-styled-components";

import { Sticky } from "components/atoms";

test("renders `Sticky` component", () => {
  const { getByTestId } = render(<Sticky data-testid="sticky-component" />);
  expect(getByTestId("sticky-component")).toBeInTheDocument();
});
test("test `Sticky` component position", () => {
  const component = create(<Sticky />).toJSON();
  expect(component).toMatchSnapshot();
  expect(component).toHaveStyleRule("position", "sticky");
});
