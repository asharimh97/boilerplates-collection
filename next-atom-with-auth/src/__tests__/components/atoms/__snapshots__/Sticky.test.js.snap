// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test \`Sticky\` component position 1`] = `
.c0 {
  position: -webkit-sticky;
  position: -webkit-sticky;
  position: sticky;
}

<div
  className=" c0"
/>
`;
